### ISSUE TITLE
[Question] Tuning minimap2 for chimeric reads

### ISSUE THREAD

**Issue by majorkazer** [WGLab/LongGF#7]
Hello,

I have some direct RNA-sequencing data generated with oxford nanopore that I am trying to process for chimeric reads. I aligned my data using the suggested parameters from minimap2, but am getting only 2-3 hits with LongGF on BAM files containing 3-10M reads. Do you have a suggestion on how to change the minimap2 parameters for maximizing mapping of chimeric reads? I was thinking of changing the -G parameter (allowed gap size). Relatedly, does LongGF look at the unmapped reads within a BAM?

minimap2 parameters:
`-ax splice -uf -k14`

Thanks!

**Comment by liuqianhn** [comment]
@majorkazer I do not have the experience to maximize mapping of chimeric reads with minimap2, since I use the commends you used for the alignment. 
1. LongGF does not check unmapped reads.
2. Do your sort BAM by name (not by position in default) before using LongGF? 
If you can share the commands you used for mapping and sorting BAM and LongGF commands together with output log, it will be helpful to investigate the issue you have.

**Comment by majorkazer** [comment]
Thanks for getting back to me so quickly! I also posted a question on the minimap2 git about this, but haven't really gotten a satisfying answer other than that the chimeric reads are specified in the SAM... Are there any considerations for the input GTF in terms of annotation of transcripts? Does LongGF work off CDS, exons, transcripts, etc?

I did sort my BAM by name before using LongGF. Here is my workflow:
```
minimap2 -ax splice -uf -k14 mm10.fa nanopore.fastq > nanopore.sam

samtools view -S -b nanopore.sam > nanopore.bam
samtools sort -n nanopore.bam -o nanopore.name.sorted.bam

LongGF nanopore.name.sorted.bam mm10.gtf 100 50 100 > LongGF.log
```
I also attached an example log.
[LongGF_R20-2535v1.log](https://github.com/WGLab/LongGF/files/6281637/LongGF_R20-2535v1.log)

**Comment by liuqianhn** [comment]
@majorkazer I checked the full log file. Could you please try the LongGF command below
```
LongGF nanopore.name.sorted.bam mm10.gtf 100 50 100 2 0 1 0 > LongGF.log
```

With the command above, the threshold for supporting reads is re-set to 1 (default is 2), and all genes are considered (you need to filter pseudogene by yourself).

Please note that I assume that you do not install LongGF from conda since I tried to merge my latest revision to conda but got no response from conda yet. Thus, the conda version do not include my latest revision.

Another potential issue is that: there are much less multiple mapped reads. You can show the potential coverage to investigate this possibility.

**Comment by majorkazer** [comment]
@liuqianhn Thanks for getting back to me.

I actually am using the conda version - I was having trouble installing with g++; the build was failing due to some issue with finding htslib links. Should I try to get the git version installed before continuing with your suggested command?

I am also curious about the small number of mapped reads. I have seen in some publications that the minimap2 command is run with `secondary=no`, but I think that just affects what is written to the output sam and doesn't actually change the alignment.

**Comment by majorkazer** [comment]
I was able to get the git clone to compile successfully and run the command as you suggested - this did capture fusion events with only one supporting read.

I also reduced the allowed intron size parameter `-G 200k` (default) in my minimap2 alignment and prevented printing of secondary alignments. These bams yielded more reads with multiple mappings and increased the number of fusion events detected using LongGF. See below for my minimap2 call.

`minimap2 -ax splice -uf -k14 -G 50k --secondary=no mm10.fa nanopore.fastq > nanopore.sam`
